### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by schmidsi**
This are my first steps into making the Engine Typescript ready. Linting goes through and typechecking in VS code worked for my testfile `core-worker/plugins/heartbeat.ts`.  But somehow, the tests do not run at all on my computer.

It is only a draft because the tests didn't went through yet.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -28,6 +34,22 @@
...
-    ]
-  }
+    ],
+    "no-unused-vars": "off",
+    "@typescript-eslint/no-unused-vars": "warn",
+    "@typescript-eslint/no-empty-function": "warn",
+    "@typescript-eslint/explicit-module-boundary-types": "off",
+    "@typescript-eslint/no-var-requires": "off"
+  },
+  "overrides": [
```
Comment (schmidsi) [inline .eslintrc:44]: This is a cool pattern that I discovered: We can introduce more strict linting rules for .ts files. So the code quality increases by migrating to Typescript not only through the types but also through the linting. But the old code just works as before.

File: packages/api/api.js [hunk 2]
```diff
@@ -4,10 +4,10 @@ import createGraphQLServer from './createGraphQLServer';
 import createBulkImportServer from './createBulkImportServer';
 import { configureRoles } from './roles';
 
-export hashPassword from './hashPassword';
-export getCart from './getCart';
-export evaluateContext from './evaluateContext';
-export filterContext from './filterContext';
+export { default as hashPassword } from './hashPassword';
```
Comment (schmidsi) [inline packages/api/api.js:7]: This is needed for the Typescript parser

File: packages/core-worker/plugins/heartbeat.ts [hunk 3]
```diff
@@ -1,15 +1,25 @@
...
 
-class Heartbeat extends WorkerPlugin {
+type Arg = {
+  wait?: number;
+  fails?: boolean;
+};
+
+type Result = Arg;
+
+class Heartbeat extends WorkerPlugin<Arg, Result> {
```
Comment (schmidsi) [inline packages/core-worker/plugins/heartbeat.ts:22]: Example of a correctly typed worker-plugin

File: packages/core-worker/types.d.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+declare module 'meteor/unchained:core-worker' {
```
Comment (schmidsi) [inline packages/core-worker/types.d.ts:1]: How to type `import * from 'meteor/unchained:core-worker';` imports

**Comment by schmidsi** [comment]
@pozylon Maybe you can merge these changes to https://github.com/unchainedshop/unchained/tree/ts-ausflug

File: .eslintrc (hunk 1 above)
Comment (pozylon) [reply .eslintrc:44]: Nice

**Comment by pozylon** [comment]
I'll merge it into the bookmark PoC then fix tests

**Comment by harryadel** [comment]
Meteor Roles PR contains some typescript setup, maybe we can use it as a starting point to avoid replicating work.

https://github.com/unchainedshop/unchained/pull/253

**Comment by pozylon** [comment]
Part of bookmarks poc now so we can discuss future direction there

**Pull request closed without merging**